src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,125 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private final StorageDao storageDao = new StorageDaoImpl();
+    private final User correctUser = new User();
+    private final User sameLogin1User = new User();
+    private final User sameLogin2User = new User();
+    private final User shortLoginUser = new User();
+    private final User shortPasswordUser = new User();
+    private final User tooYoungUser = new User();
+    private final User shadowUser = new User();
+    private final User nullLoginUser = new User();
+    private final User nullPasswordUser = new User();
+    private final User nullAgeUser = new User();
+
+    @BeforeEach
+    void setUp() {
+        correctUser.setLogin("CorrectUserLogin");
+        correctUser.setPassword("CorrectUserPassword");

```

Comment:
it doesn't look nice
either make all these users local variables, specific to each test
or have one correctuser field and reuse it across tests, setting user fields as you need (for example, age to 17)